forecasttools/data.py

```diff
@@ -0,0 +1,478 @@
+                smooth=False,
+                fill_kwargs={"alpha": 0.6, "label": "50% Credible"},
+                ax=axes,
+            )
+            axes.plot(
+                X,
+                y,
+                marker="o",
+                color="black",
+                linewidth=1.0,
+                markersize=3.0,
+                label="Observed",
+            )
+            median_ts = y_data.median(dim=["chain", "draw"])
+            axes.plot(x_data, median_ts, color="blue", label="Median")
+            axes.legend()
+            axes.axvline(last, color="black", linestyle="--")
+            axes.set_title(
+                f"Hospital Admissions ({state}, {start_date}-{end_date})",
+                fontsize=15,
+            )
+            axes.set_xlabel("Time", fontsize=20)
+            axes.set_ylabel("Hospital Admissions", fontsize=20)
+            plt.show()
+
+
+def read_example_flusight_submission(
+    save_directory: str,
+    file_save_name: str,
+    create_save_directory: bool = False,
+) -> None:
+    """
+    Parameters
+    ----------
+    save_directory
+        The directory to save the outputted csv in.
+    file_save_name
+        The name of the file to save.
+    create_save_directory
+        Whether to make the save directory, if it
+        does not exist.
+    """
+    # check if the save directory exists
+    if not os.path.exists(save_directory):
+        if create_save_directory:
+            os.makedirs(save_directory)
+            print(f"Directory {save_directory} created.")
+        else:
+            raise FileNotFoundError(
+                f"The directory {save_directory} does not exist."
+            )
+    # check if save directory is actual directory
+    if not os.path.isdir(save_directory):
+        raise NotADirectoryError(
+            f"The path {save_directory} is not a directory."
+        )
+    # check if the save file already exists
+    file_save_path = os.path.join(save_directory, file_save_name)
+    if os.path.exists(file_save_path):
+        raise FileExistsError(f"The file {file_save_path} already exists.")
+    else:
+        # check if the FluSight example url is still valid
+        url = "https://raw.githubusercontent.com/cdcepi/FluSight-forecast-hub/main/model-output/cfa-flumech/2023-10-14-cfa-flumech.csv"
+        try:
+            response = requests.get(url, timeout=10)
+            if response.status_code == 200:
+                print(f"FluSight example url is valid: {url}")
+            else:
+                raise ValueError(
+                    f"FluSight example url bad status: {response.status_code}"
+                )
+        except requests.exceptions.RequestException as e:
+            raise ValueError(f"Failed to access FluSight example url: {e}")
+        # read csv from URL, convert to polars
+        submission_df = pl.read_csv(url, infer_schema_length=7500)
+        # save the dataframe
+        submission_df.write_csv(file_save_path)
```

Comment:
This seems unnecessary. Can't we just download the csv to disk without reading it into polars?

Reply:
Yes, though not sure if a future iteration might have modification options available, in which case the `polars` already being present might be nice.